tyssue/core/sheet.py

```diff
@@ -91,6 +91,102 @@ def get_neighborhood(self, face, order):
 
         return neighbors.reset_index(drop=True).loc[1:]
 
+    def sheet_extract(self, key_word_column, coords=['x', 'y', 'z']):
+        """ Extract a new sheet from the embryo sheet
+        that correspond to a key word that define a face.
+        Parameters
+        ----------
+        sheet: a :class:Sheet object
+        coords
+        Returns
+        -------
+        sheet_fold_patch_extract :
+            subsheet corresponding to the fold patch area.
+
+        """
+        x, y, z = coords
+        datasets = {}
+
+        datasets['face'] = self.face_df[
+            self.face_df[key_word_column] == True].copy()
+        datasets['edge'] = self.edge_df[self.edge_df['face'].isin(
+            datasets['face'].index)].copy()
+        datasets['vert'] = self.vert_df[self.vert_df['srce_o'].isin(
+            datasets['edge']['srce_o'])].copy()
+
+        # reasign index
+        old_index = list(datasets['vert'].index)
+        datasets['vert'].reset_index(drop=True, inplace=True)
+
+        # Dictionnary creation
+        dic = {}
+        for i in range(len(old_index)):
+            dic[old_index[i]] = datasets['vert'].index[i]
+
+        datasets['edge']['srce'].replace(dic, inplace=True)
+        datasets['edge']['trgt'].replace(dic, inplace=True)
+        datasets['edge']['srce_o'].replace(dic, inplace=True)
+        datasets['edge']['trgt_o'].replace(dic, inplace=True)
+
+        subsheet = Sheet('subsheet', datasets, self.specs)
+        return (subsheet)
+
+
+    def sheet_extract_coordinate(self, xmin, xmax, ymin, ymax, zmin, zmax,
+                                 coords=['x', 'y', 'z']):
+        """ Extract a new sheet from the embryo sheet
+        that correspond to boundary coordinate
+        define by the user.
+        Parameters
+        ----------
+        sheet: a :class:Sheet object
+        xmin, xmax : boundary
+        ymin, ymax : boundary
+        zmin, zmax : boundary
+        coords
+
+        Returns
+        -------
+        sheet_extract :
+            subsheet
+
+        """
+        x, y, z = coords
+        datasets = {}
+
```

Comment:
It might be possible to make a more generic version of the function that would work in 2D and 3D by passing a bounding box like [[x_min, x_max], [y_min, y_max], [z_min, z_max]] as an argument.

I remember now I implemented something approaching here:
https://github.com/DamCB/tyssue/blob//tyssue/core/objects.py#L594

Maybe it's possible to factor the bounding box code to get what's in and what's out of it, for both `cut_out` and this method. You can then generate a face mask from it and call the `sheet_extract` function, thus avoiding duplicate code (yay less code ;-D )